src/matchers/toHaveQueryParam.ts

```diff
@@ -0,0 +1,40 @@
+import { decodeQueryParams, QueryParamConfig } from 'serialize-query-params'
+import { parse } from 'query-string'
+import isEqual from 'lodash.isequal'
+
+export interface ExpectedQueryParam {
+  name: string;
+  type: QueryParamConfig<any, any>
+  value: any;
+}
+
+export const toHaveQueryParam = (
+  locationSearch: Location['search'],
+  { name, value, type }: ExpectedQueryParam
+) => {
+  const encodedValue = parse(locationSearch)[name]
+
+  if (!encodedValue) {
+    return {
+      pass: false,
+      message: () => `${name} wasn't found in ${locationSearch}`
+    }
+  }
+
+  const decodedValue = decodeQueryParams(
+    { [name]: type },
+    { [name]: encodedValue }
+  )[name]
+
+  const formatExpectedQueryParamValue = typeof value === 'object'
+    ? JSON.stringify(value)
+    : value
+  const formatReceivedQueryParamValue = typeof decodedValue === 'object'
```

Comment:
Still not confident with this logic to display the values in the error message. This is going to be improved in the next PR. 

Maybe, in this case, it could be extracted to a separate function in the same file... but I don't know if converting values to string is always the best case.

Reply:
Improved at https://github.com/LauraBeatris/react-router-testing-utils/pull/4